Add interceptors for stream requests

florian-wagner-frequenz opened the pull request:
This adds authentication and signing interceptors for stream requests. Specifically for those where the client sends a single message and receives a stream.

llucax reviewed (approved):
LGTM. 2 very minor and optional comments:

* Split the `erros` -> `errors` typo fix into a separate commit
* Put the common code into a *mixin class* (like `SigningInterceptorMixin`) and then inherit from both the interceptor and the mixin. This can save a couple of extra lines (basically the `__init__()` method). Mostly FYI in case you are not familiar with this option in Python, the code is so simple that for me it is still perfectly fine too keep it as it is.

florian-wagner-frequenz committed:
Fix Typo

Signed-off-by: Florian Wagner <florian.wagner@frequenz.com>

florian-wagner-frequenz committed:
Add interceptors for stream requests

This adds authentication and signing interceptors for stream requests.
Specifically for those where the client sends a single message and
receives a stream.

Signed-off-by: Florian Wagner <florian.wagner@frequenz.com>

florian-wagner-frequenz commented on src/frequenz/client/base/signing.py:
```diff
@@ -9,17 +9,65 @@
+    hmac_obj.update(nonce_bytes)
+
+    # Once again, gRPC is mistyped
+    hmac_obj.update(client_call_details.method.split(b"/")[-1])  # type: ignore[arg-type]
```
This is mistyped in grpc

florian-wagner-frequenz commented:
I decided not to do the mixin, but split out the error.
If we decide to do the other two classes (StreamUnary, StreamStream), I will do the mixin.
